polkadot/node/subsystem-bench/src/subsystem-bench.rs

```diff
@@ -76,12 +76,31 @@ struct BenchCli {
 	/// Maximum remote peer latency in milliseconds [0-5000].
 	pub peer_max_latency: Option<u64>,
 
+	#[clap(long, default_value_t = false)]
+	/// Enable CPU Profiling
+	pub profile: bool,
+
 	#[command(subcommand)]
 	pub objective: cli::TestObjective,
 }
 
 impl BenchCli {
 	fn launch(self) -> eyre::Result<()> {
+		use pyroscope::PyroscopeAgent;
+		use pyroscope_pprofrs::{pprof_backend, PprofConfig};
+
+		// Pyroscope must be running on port 4040
+		// See https://grafana.com/docs/pyroscope/latest/get-started/#download-and-configure-pyroscope
+		let agent_running = if self.profile {
+			let agent = PyroscopeAgent::builder("http://localhost:4040", "subsystem-bench")
```

Comment:
To not couple the code to a specific deployment configuration this needs to be a configurable parameter of the CLI.

Reply:
Ok, but with default to fast run it like we do with Prometheus